.github/scripts/changed-adapters.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash -e
+
+# If BUILD_ALL is true, then we want to generate lists with all packages regardless of changes
+if [[ $BUILD_ALL = true ]]; then
+  PACKAGE_LIST=$(yarn workspaces list -R --json)
+else
+  PACKAGE_LIST=$(yarn workspaces list -R --json --since=$UPSTREAM_BRANCH)
+fi
+
+# The yarn command used above will give us a list of the packages that have changed (including changes by dependencies)
+CHANGED_PACKAGES=$(
+  echo $PACKAGE_LIST \
+  | jq -cs '.' \
+  | jq -cr '[.[] | select(.location | startswith("packages"))]' \
+)
+
+# Add versions to the packages
+CHANGED_PACKAGES=$(
+  for package in $(echo $CHANGED_PACKAGES | jq -c '.[]'); do
+    location=$(echo $package | jq -r '.location')
+    version=$(jq '.version' $location/package.json)
+    echo $package | jq -c ".version = $version"
+  done | jq -cs '.'
+)
+
+# Build a list to use with matrix strategies
+CHANGED_ADAPTERS=$(
+  echo $CHANGED_PACKAGES \
+  | jq -cr '{
+      adapter: [
+        .[]
+        | select(
+          .location
+          | startswith("packages/sources") or startswith("packages/composites") or startswith("packages/targets"))
```

Comment:
Maybe we can look into decommissioning the `targets` EAs since they aren't really being developed at this point. Not for this PR, but for future ref.

Reply:
Agreed, there's an ongoing plan to clean a lot of these EAs but it might take a while to make sure it's proper to do so